tests/examples/simple.cpp

```diff
@@ -20,6 +20,39 @@ struct Error final {
   std::string what;
 };
 
+template <typename T> struct immovable_Value final {
```

Comment:
Was it meant to be `immovable_Value` and `immovable_Fn`? Why not `immovable_value` and `immovable_fn`?

Reply:
Or if this is supposed to represent a user type then perhaps `ImmovableValue` since we already have `Error` in the mix?